### PR TITLE
PERF-3752 Add new time-series multi delete workload

### PR DESCRIPTION
**Pull request by YuhongZhang98**
patch:
https://spruce.mongodb.com/version/6413d3e80ae6060dd9baea71/tasks?page=0&sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&taskName=delete

### CODE REVIEW

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [inline src/workloads/execution/TimeSeriesRangeDelete.yml:105]: I can merge this in after [SERVER-74899](https://jira.mongodb.org/browse/SERVER-74899) and enable the variant.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml]: Updated.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml]: Yes, from our PM the only use case mentioned in https://docs.google.com/document/d/1O8k-8rouefH69htK-zoj3tbmcfw0ivfaZqfS8xrw4pU/edit#heading=h.pmdx0igul501 is the customer might delete some 1-hour range buckets. I wanted to cover both full-bucket and partial-bucket deletions so I extended it to span across three buckets.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml]: Removed.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml:101]: Done.

**Review by yun-soo — COMMENTED** [review]
I'm a bit confused by data insert. Left a few questions on that and could you clarify data insert?

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml:2]: Yes, from other genny workloads, I saw `server-execution` for workloads our team maintains. I didn't see any `storage-execution`.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml:28]: Split to two actors after the refactoring.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml:59]: I was trying to use `*` but it didn't work. Glad we can use underscore instead!

We have 1000 devices, each has 100 buckets, and each bucket has 100 measurements.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml:6]: That is correct Fausto. The updated comments are in the header. It seems like people didn't see it. Maybe need to pull it so that it will show up in vscode?

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (Faustoleyva54) [inline src/workloads/execution/TimeSeriesRangeDelete.yml:59]: hmmm, based of of my previous comment it looks like this count might have to be 10_000 ? or the docCount should be 1_000_000? But I might also be a tad confused 😅

Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml:59]: What this line does is it generates 1000 entries for each timestamp. So we insert timestamp1 for all 1000 devices first, and then timestamp2, and so on.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (yun-soo) [reply src/workloads/execution/TimeSeriesRangeDelete.yml]: Just found that we have `^NumExpr`. Here's an example for it.

https://github.com/mongodb/genny/blob/master/src/workloads/docs/HelloWorld.yml/#L33-#L35

Let's define three variables like `nSensors`, `nBucketsPerSensor`, `nMeasurementsPerBucket` or somethings similar and use them instead of `10_000_000`.

**Review by yun-soo — COMMENTED** [review]
Thanks for updating the PR, Yuhong! Hope we can go an extra mile to help readers easily understand the insert process.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (Faustoleyva54) [reply src/workloads/execution/TimeSeriesRangeDelete.yml:59]: I see. So we have something like this occurring: 

```
for docId in range(0, DocumentCount=10_000_000):
    for ts in range(0, *Initial ts + step x 1000*, step=36000):
        for meta in range(0, ofLength=1000):
            coll.insert({_id, docId, t: ts, "sensorId": meta,...})
```

**Review by Faustoleyva54 — APPROVED** [review]
LGTM % comment explaining granularity and 10_000_000 doc count.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml]: done.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml]: Added more comments to explain the data generation and granularity setting in collection creation.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml:6]: I was referring to https://github.com/mongodb/genny/pull/843/files#diff-b789006c21bde7a9b14d1f5d144373ad4adcff0348dca29fb365b1d21994f6f5R9-R15. But I added more inline comments to explain how specific data generation was done.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml]: Added inline comments.

**Review by awagenmaker — DISMISSED** [review]
LGTM! Thanks for adding more comments about the data generation.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (yun-soo) [inline src/workloads/execution/TimeSeriesRangeDelete.yml:16]: [optional]
```suggestion

  We have 1000 independent sensors which will have 100 buckets, and each bucket has 100 measurements.

  The buckets have the meta field, sensorId, with values from 0 to 999, each of which has 100 measurements with
  timestamps within an hour with 36-second intervals. For example, in a bucket, the measurements
  look like:
  {t: 2023-01-01T5:00:00, m: 42, ...},
  {t: 2023-01-01T5:00:36, m: 42, ...},
  ...
  {t: 2023-01-01T5:59:24, m: 42, ...}

```

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (yun-soo) [inline src/workloads/execution/TimeSeriesRangeDelete.yml:69]: [optional] Following comment or something similar along the line.
```suggestion
              # 1hour time span per a bucket (== 3600 seconds) / 36000 millis (== 36 seconds) increment per each measurement == 100 measurements per a bucket
              nMeasurementsPerBucket: *nMeasurementsPerBucket
```

**Review by yun-soo — APPROVED** [review]
Thanks for updating the PR and LGTM! There are two optional comments which are up to you. I'm pretty happy with the current PR.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml]: done.

File: src/workloads/execution/TimeSeriesRangeDelete.yml
Comment (YuhongZhang98) [reply src/workloads/execution/TimeSeriesRangeDelete.yml:71]: done.

**Comment by YuhongZhang98** [comment]
Waiting for sharding support for multi deletes to merge in. No further actions needed by then.

**Comment by YuhongZhang98** [comment]
Rerequesting an approval to unblock the github merge

**Review by yun-soo — APPROVED** [review]

**Pull request merged**
